### PR TITLE
utils: aws client logging

### PR DESCRIPTION
**Pull request by bobheadxi**
:tickets: **Ticket(s)**: n/a

---

## :construction_worker: Changes

Adds optional logger argument for our aws config constructor to set a logger.

Currently defaults to errors only. `AWS_DEBUG=true` enables everything. The API does not allow a lot of configuration: https://docs.aws.amazon.com/sdk-for-go/api/aws/#LogLevelType

## :flashlight: Testing Instructions

run core with `AWS_DEBUG=true`

### CODE REVIEW

File: utils/aws.go [hunk 1]
```diff
@@ -21,12 +32,32 @@ func AWSConfig(dev bool) (cfg *aws.Config) {
...
-		if os.Getenv("AWS_DEBUG") == "true" {
-			cfg.LogLevel = aws.LogLevel(aws.LogDebug)
-		}
 	} else {
-		// todo: production aws setup
+		// todo: more granular production aws setup
+		cfg = aws.NewConfig()
+	}
+
+	if os.Getenv(AWSDebugEnv) == "true" {
```
Comment (srijonsaha) [inline utils/aws.go]: is this how logging will be toggled in code? like everything you want to log has to like this?
```go
if os.Getenv(AWSDebugEnv) == "true" {
    // log stuff
}
```

**Comment by srijonsaha** [comment]
Is this automatically going to log all DynamoDB stuff if AWS_DEBUG is on?

File: utils/aws.go (hunk 1 above)
Comment (bobheadxi) [reply utils/aws.go]: Preferably no - this function only creates the AWS config, so what the env variables toggles here is whether we use the debug mode in the AWS API

Comment (bobheadxi) [reply utils/aws.go]: The debug mode enables verbose logging to stdout by default, this PR adds the ability for it to send output to the logger instead

**Comment by bobheadxi** [comment]
@srijonsaha yep!

**Pull request merged**